File: frontend/src/microservices/content/objects/ExploreSuggestionPlannerResponse.ts
// ExploreSuggestionPlannerResponse：content 域单条探索推荐返回对象。

export type ExploreSuggestionPlannerResponse = {
  resourceType: string
  value: string
  title: string
  subtitle: string
}
export const exploreSuggestionPlannerResponseFromJson = (json: string): ExploreSuggestionPlannerResponse =>
  JSON.parse(json) as ExploreSuggestionPlannerResponse
export const exploreSuggestionPlannerResponseToJson = (value: ExploreSuggestionPlannerResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/content/objects/ExploreSuggestionPlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ExploreSuggestionPlannerResponse(resourceType: String, value: String, title: String, subtitle: String)
object ExploreSuggestionPlannerResponse:
  given sourceEncoder: Encoder[ExploreSuggestionPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ExploreSuggestionPlannerResponse] = deriveDecoder
